client/components/domains/domain-suggestion-flag/index.jsx

```diff
@@ -0,0 +1,37 @@
+/**
+ * External dependencies
+ */
+var React = require( 'react' );
+
+/**
+ * Internal dependencies
+ */
+var Flag = require( 'components/flag' );
+
+const DomainSuggestionFlag = React.createClass( {
+	propTypes: {
+		domain: React.PropTypes.string
```

Comment:
What's the use case when the domain is null? :) I think the safest option would be to make this prop to be required. This would simplify the `render` function too.